src/coreComponents/dataRepository/DataContext.hpp

```diff
@@ -0,0 +1,261 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file DataContext.hpp
+ */
+
+#ifndef GEOS_DATAREPOSITORY_DATACONTEXT_HPP_
+#define GEOS_DATAREPOSITORY_DATACONTEXT_HPP_
+
+#include "common/DataTypes.hpp"
+#include "common/Logger.hpp"
+#include "xmlWrapper.hpp"
+#include "common/Format.hpp"
+#include "Group.hpp"
+#include "WrapperBase.hpp"
```

Comment:
I think we are in the same situation that `Wrapper` and `Group` are: we need to access the `DataContext` from a `Group`, and we need a `Group` to initialize the `DataContext`. That's why I used the same technique to solve it (the forward declaration).

Another way to solve it would be to move the declaration of GroupContext & WrapperContext in a dedicated `GroupContext.hpp` file which would include `Group.hpp`. I would also construct `DataFileContext` directly from strings produced from outside of the `DataFileContext` class.